src/PolyhedralGeometry/Polyhedron/constructors.jl

```diff
@@ -68,9 +68,9 @@ julia> vertices(P)
  [0, 0]
 ```
 """
-Polyhedron{T}(A::Union{Oscar.MatElem,AbstractMatrix}, b) where T<:scalar_types = Polyhedron{T}((A, b))
+Polyhedron{T}(A::MatrixUnion, b) where T<:scalar_types = Polyhedron{T}((A, b))
 
-function Polyhedron{T}(I::Union{Nothing, SubObjectIterator, Tuple{<:Union{Oscar.MatElem, AbstractMatrix}, Any}}, E::Union{Nothing, SubObjectIterator, Tuple{<:Union{Oscar.MatElem, AbstractMatrix}, Any}} = nothing) where T<:scalar_types
+function Polyhedron{T}(I::Union{Nothing, TupleUnion}, E::Union{Nothing, TupleUnion} = nothing) where T<:scalar_types
```

Comment:
This is the only place using `TupleUnion` right now. Do you plan to use it in more places?

This constructor does not have a docstring, so it is not really visible to the user. 

My personal two cents: I found this constructor quite confusing when I tried to untangle and understand how the various `Polyhedron` constructor interact. I am not quite sure why it takes tuples anyway; what is the second part in those tuples doing anyway?